### DISCUSSION TITLE
Hardware support

### DISCUSSION THREAD

**Discussion by Alden-G878** [k3s-io/k3s#4989] [Q&A]
I know that special Nvidia software is needed to use CUDA GPUs with k3s. I intend to make a k3s cluster using pine64's SoEdge modules, which have a Rockchip RK1808 SoC that has a 3.0 TOPS NPU. Would similar work have to be done for k3s to recognize this NPU like the CUDA GPUs?

**Comment by brandond** [comment]
K3s doesn't recognize anything about the hardware itself. The only nvidia-specific integration we have is looking for and enabling the nvidia container runtime, if installed.

**Reply by Alden-G878** [reply]
In which case would I need a similar container runtime for k3s to work with a NPU?

**Reply by brandond** [reply]
I'm not sure. That would be up to the specific hardware vendor. The nvidia container runtime mostly just adds support for exposing the correct subset of hardware to the container, in environments where it may be desirable to allocate specific resources to pods; see https://github.com/NVIDIA/nvidia-container-runtime#nvidia-container-runtime - for a smaller device like yours its possible that you don't need to do anything special.

**Reply by Alden-G878** [reply]
Thanks!
